surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/logger/SurefireProviderLoggerFactory.java

```diff
@@ -25,40 +32,69 @@ public Logger getLogger() {
 
         private static final String PREFIX = "%s: Smart Testing Extension - ";
 
-        private ConsoleLogger consoleLogger;
+        private Object consoleLogger;
+        private Method logMethod;
 
-        private SurefireProviderLogger(ConsoleLogger consoleLogger) {
+        private SurefireProviderLogger(Object consoleLogger) {
             this.consoleLogger = consoleLogger;
+            Optional<Method> method = Arrays.stream(consoleLogger.getClass().getMethods())
+                .filter(this::isPublicConsoleLogMethod)
+                .findFirst();
+
+            if (method.isPresent()) {
+                logMethod = method.get();
+            } else {
+                new DefaultLoggerFactory()
```

Comment:
Is it safe to continue having `logMethod` null?

Reply:
This is managed here: https://github.com/arquillian/smart-testing/pull/223/files#diff-72e56dc4b88e2415a66df5915b9e4b6cR76